chainerx_cc/chainerx/stack_vector.h

```diff
@@ -194,4 +195,33 @@ class StackVector {
     BaseContainer d_{};
 };
 
+// Formatter to print StackVector containing integral elements as e.g. '[]' or '[1, 2, 3]'.
+template <typename T, stack_vector_detail::size_type N>
+class DimArgFormatter {
+public:
+    static_assert(std::is_integral<T>::value, "DimArgFormatter requires intergral elements.");
+
+    explicit DimArgFormatter(const StackVector<T, N>& stack_vector) : stack_vector_{stack_vector} {}
+
+    void Print(std::ostream& os) const {
```

Comment:
Make this function private, and make `operator<<` friend.

Reply:
Fixed.